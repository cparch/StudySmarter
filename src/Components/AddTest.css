.addTestHeaderImage {
  background-image: url('../Images/scantronImg.jpg');
  height: 200px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
}

.addTestForm select {margin-bottom: 20%;}

@media(min-width: 901px) {
  .addTestForm select {margin-bottom: 12%;}
}

@media(min-width: 1250px) {
  .addTestForm select {margin-bottom: 9%;}
}
