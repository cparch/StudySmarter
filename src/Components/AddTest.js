import React from 'react';

const AddTest = (props) => {

  return(
    <div>
      <form onSubmit={props.AddTestSubmitBtnHandler}>
        <label>
            <h2>
              Class Name:
            </h2>
            
            <select name="selectedClass" onChange={props.FormHandler}> 
              {props.entireState.classes.map((classes, idx) => (
                <option key={classes.classTitle} value={idx}>
                  {classes.classTitle}
                </option>
              ))}
            </select>
        </label>
        <br/>
        <br/>
        <label>
          Add new Test To Study For:
          <input
            value={props.entireState.testNameToAdd} 
            onChange={props.FormHandler} 
            type='text' 
            placeholder='Chapter 1 test' 
            name='testNameToAdd'
          />
        </label>
        <br/>
        <br/>
        <input type="submit" value="Submit" />
      </form>
    </div>
  )
};

export default AddTest;