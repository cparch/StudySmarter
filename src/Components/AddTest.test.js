import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import AddTest from './AddTest';
import moment from 'moment';

 
const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

 
  beforeEach(() => {
    store = mockStore({
      FormHandlerReducer : {
        selectedClass: 0,
        selectedTest: 0,
        SelectedStartTimeValue: '',
        SelectedEndTimeValue: '',
        notes: '',
        classNameToAdd: '',
        testNameToAdd: '',
        gradeInput: '',
        showSubmitConfirmation: false,
        startTimeValue: moment(),
        endTimeValue: moment(),
      },
      classesReducer: {
        byID : {
          "class0": {
            id: "class0",
            classTitle: 'PsychologyRedux'
          },
          "class1": {
            id: "class1",
            classTitle: 'HistoryRedux',
          } 
        },
        allID: ["class0", "class1"]
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <AddTest />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 
  it('should dispatch an action on button click', () => {
 
  });
});