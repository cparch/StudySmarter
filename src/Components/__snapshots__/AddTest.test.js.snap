// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <div
    className="addTestHeaderImage"
  />
  <div
    className="mainContainer"
  >
    <h1>
      Add Test
    </h1>
    <form
      className="addTestForm"
      onSubmit={[Function]}
    >
      <label>
        <select
          className="select"
          name="selectedClass"
          onChange={[Function]}
        >
          <option>
            Class Name:
          </option>
          <option
            value="class0"
          >
            PsychologyRedux
          </option>
          <option
            value="class1"
          >
            HistoryRedux
          </option>
        </select>
      </label>
      <br />
      <div
        class="page"
      >
        <label
          class="field a-field a-field_a1"
        >
          <input
            class="field__input a-field__input"
            name="testNameToAdd"
            onChange={[Function]}
            placeholder="Chapter 1 test"
            required={true}
            value=""
          />
          <span
            class="a-field__label-wrap"
          >
            <span
              class="a-field__label"
            >
              Test I'm Studying For
              :
            </span>
          </span>
        </label>
      </div>
      <br />
      <input
        className="button"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
